frame/benchmarking/src/analysis.rs

```diff
@@ -49,7 +50,10 @@ fn mul_1000_into_u128(value: f64) -> u128 {
 impl BenchmarkSelector {
 	fn scale_and_cast_weight(self, value: f64, round_up: bool) -> u128 {
 		if let BenchmarkSelector::ExtrinsicTime = self {
-			mul_1000_into_u128(value)
+			// We add a very slight bias here to counteract the numerical imprecision of the linear
+			// regression where due to rounding issues it can emit a number like `2999999.999999998`
+			// which we most certainly always want to round up instead of truncating.
+			mul_1000_into_u128(value + 0.000_000_005)
```

Comment:
Can you not use `ceil()` somewhere here to keep the logic simple?

Reply:
`ceil` will round the whole thing, e.g. let's assume you have a value of `10.599999999` - using a `ceil` would round it to `11`, while with this it will round it to `10.6`, so the behavior would be quite different